netty/src/main/java/io/grpc/netty/ProtocolNegotiators.java

```diff
@@ -103,6 +109,47 @@ public AsciiString scheme() {
   }
 
   /**
+   * Create a server plaintextUpgrade handler for gRPC.
+   */
+  public static ProtocolNegotiator serverPlaintextUpgrade() {
+    return new ProtocolNegotiator() {
+      public Handler newHandler(final GrpcHttp2ConnectionHandler handler) {
+        class PlaintextUpgradeHandler extends ChannelHandlerAdapter implements Handler {
+          PlaintextUpgradeHandler() {
+          }
+
+          public void handlerAdded(ChannelHandlerContext ctx) throws Exception {
+
+            final HttpServerUpgradeHandler.UpgradeCodecFactory upgradeCodecFactory = new HttpServerUpgradeHandler.UpgradeCodecFactory() {
```

Comment:
We choose not to use `final` for local variables, unless necessary for closures. It looks like this `final` can be removed.